examples/Example/src/prot_snap/index.tsx

```diff
@@ -0,0 +1,347 @@
+        </Animated.View>
+      </GestureMonitor>
+
+      <GestureMonitor gesture={panGestureHandler}>
+        <Animated.View style={styles.buttonContainer}>
+          <FilterCarousel filters={filters} selected={filter} />
+          <GestureMonitor gesture={buttonGestureHandler}>
+            <CaptureButton progress={1 - remainingTime / MAX_VIDEO_DURATION} />
+          </GestureMonitor>
+        </Animated.View>
+      </GestureMonitor>
+    </Animated.View>
+  );
+}
+
+const styles = StyleSheet.create({
+  container: {
+    width: '100%',
+    height: '100%',
+    alignSelf: 'center',
+    backgroundColor: 'white',
+  },
+  home: {
+    width: '100%',
+    height: '100%',
+    alignSelf: 'center',
+    opacity: 0.5,
+  },
+  buttonContainer: {
+    width: '100%',
+    height: 100,
+    position: 'absolute',
+    bottom: 50,
+    zIndex: 10,
+  },
+  shutterContainer: {
+    width: 100,
+    height: 100,
+    alignSelf: 'center',
+  },
+  box: {
+    alignSelf: 'center',
+    margin: 100,
+    width: 200,
+    height: 200,
+    backgroundColor: 'black',
+  },
+});
+
+function FilterCarousel(props) {
+  return (
+    <Animated.View style={filterCarouselStyles.container}>
+      {props.filters.map((filter) => {
+        return (
+          <Filter key={filter} filter={filter} selected={props.selected} />
+        );
+      })}
+    </Animated.View>
+  );
+}
+
+function Filter(props) {
+  const window = useWindowDimensions();
+  let style = useAnimatedStyle(() => {
+    return {
+      transform: [
+        {
+          translateX: window.width * 0.5 - 35 - 15 - 100 * props.selected.value,
+        },
+      ],
+    };
+  });
+
+  return (
+    <Animated.View
+      style={[
+        filterCarouselStyles.filter,
+        style,
+        { backgroundColor: props.filter },
+      ]}
+    />
+  );
+}
+
+const filterCarouselStyles = StyleSheet.create({
+  container: {
+    width: '100%',
+    height: 100,
+    alignSelf: 'center',
+    position: 'absolute',
+    flexDirection: 'row',
+  },
+  filter: {
+    width: 70,
+    height: 70,
+    borderRadius: 100,
+    alignSelf: 'flex-start',
+    margin: 15,
+    borderWidth: 0.5,
+    borderColor: 'gray',
+  },
+});
+
+function CaptureButton(props) {
+  function getOverlay(progress) {
+    if (progress > 1) {
+      progress = 1;
+    }
+
+    if (progress <= 0.5) {
+      return (
+        <Animated.View
+          style={[
+            captureButtonStyles.overlay,
+            captureButtonStyles.overlayLessThanHalf,
+            {
+              transform: [{ rotateZ: 45 + 360 * progress + 'deg' }],
+            },
+          ]}
+        />
+      );
+    } else {
+      return (
+        <Animated.View
+          style={[
+            captureButtonStyles.overlay,
+            captureButtonStyles.overlayMoreThanHalf,
+            {
+              transform: [{ rotateZ: 45 + 360 * (progress - 0.5) + 'deg' }],
+            },
+          ]}
+        />
+      );
+    }
```

Comment:
I would find it more readable if you did something like so:
```ts
    const progressBelowHalf = progress <= 0.5;
    return (
      <Animated.View
        style={[
          captureButtonStyles.overlay,
          progressBelowHalf
            ? captureButtonStyles.overlayLessThanHalf
            : captureButtonStyles.overlayMoreThanHalf,
          {
            transform: [
              {
                rotateZ:
                  45 +
                  360 * (progressBelowHalf ? progress : progress - 0.5) +
                  'deg',
              },
            ],
          },
        ]}
      />
    );
```

You can use templating as well if Reanimated supports that (``rotateZ: `${45 + 360 * (progressBelowHalf ? progress : progress - 0.5)}deg`,``).